### PR TITLE
📝  Inc. how-to undistort Neon's scene camera and gaze

### PR DESCRIPTION
**Pull request by mikelgg93**

### CODE REVIEW

File: src/neon/how-tos/advance-analysis/undistort.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+Beyond the scientific realm, the application of fisheye lens correction extends to practical domains as well. If you plan to incorporate Neon to virtual reality or augmented reality, you will need to affix the Neon module to the headset and to compute the geometrical relationship from the scene camera to the virtual reality camera. By accounting for fisheye lens distortions, you can better correlate the gaze position with the underlying geometry of the virtual environment, thus improving the accuracy of your gaze-based interactions.
+
+## How to correct for fisheye lens distortions?
+In this tutorial, we will cover how to read the provided intrinsic and extrinsic camera parameters and correct for fisheye lens distortions in Neon's scene camera. We will also show you how to undistort the gaze data, so that you can map the gaze position onto the undistorted scene camera video.
```
Comment (marc-tonsen) [inline src/neon/how-tos/advance-analysis/undistort.md:19]: The extrinsics usually describe how a camera is positioned in the world, which we don't really have, do we? As far as I can tell we are only using intrinsics.

File: src/neon/how-tos/advance-analysis/undistort.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
+---
+title: Undistorting Video and Gaze Data
+description: "Undistorting Neon's scene camera video and gaze data, using the intrinsic and extrinsic camera parameters."
+permalink: /neon/how-tos/advance-analysis/undistort/
+---
+# Correcting for the fish-eye lens distortion on Neon's scene camera
```
Comment (marc-tonsen) [inline src/neon/how-tos/advance-analysis/undistort.md:6]: I am not sure calling our lens a fish-eye lens is correct. Technically speaking I think we have "rectilinear" distortion. OpenCV has a separate model for calibrating fisheye cameras which we are not using. And finally, fish-eye distortion is usually pretty extreme, which is not the case for us.

So maybe we just call it "lens distortion" and drop the "fish-eye"?

File: src/neon/how-tos/advance-analysis/undistort.md [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+To follow this guide, you will need to have the following libraries installed on your python environment:
+
+- OpenCV for image processing and undistortion
+- Numpy for numerical data manipulation
+
+```bash
+pip install opencv-python numpy
+```
+
+### Reading from the raw file
```
Comment (marc-tonsen) [inline src/neon/how-tos/advance-analysis/undistort.md:33]: I don't love the encouragement users get to use the binary data. If we include this, we should at least mention the cloud-based version first!

File: src/neon/how-tos/advance-analysis/undistort.md [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+
+```python
+calibration = read_from_json(path)
+R = np.eye(3)
+K = np.array(calibration["camera_matrix"])
+D = np.array(calibration["distortion_coefficients"])[0]
+```
+
+## Understanding the variables
+### Identity matrix (R)
```
Comment (marc-tonsen) [inline src/neon/how-tos/advance-analysis/undistort.md:89]: What is the point of introducing this matrix? It's not used anywhere, is it?

File: src/neon/how-tos/advance-analysis/undistort.md [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+title: Undistorting Video and Gaze Data
+description: "Undistorting Neon's scene camera video and gaze data, using the intrinsic and extrinsic camera parameters."
+permalink: /neon/how-tos/advance-analysis/undistort/
+---
+# Correcting for the fish-eye lens distortion on Neon's scene camera
+One particular challenge that often arises in eye tracking setups is the presence of [fish-eye lens distortions](https://en.wikipedia.org/wiki/Fisheye_lens), also known as barrel distortions. Fish-eye lenses, with their wide-angle characteristics, allow for a broad field of view, enabling the capture of more information within a single frame. However, this advantage comes at the cost of introducing non-linear distortions. 
+
+While we account for them in Pupil Cloud, and we even give you the possibility to download the undistorted video using the [Gaze Overlay enrichment](/enrichments/gaze-overlay), this is not the case when using the [Realtime API](/neon/real-time-api) or raw data, where you may want to correct it by yourself.
+
+## Why is it important to correct for fisheye lens distortions?
```
Comment (marc-tonsen) [inline src/neon/how-tos/advance-analysis/undistort.md:11]: Showing a side-by-side of distorted vs undistorted somewhere like in the analogue how-to for undistorting PI data would be great!

**Comment by mikelgg93** [comment]
@marc-tonsen it was a 🚧 WIP but thanks for the feedback!

File: src/neon/how-tos/advance-analysis/undistort.md (hunk 1 above)
Comment (mikelgg93) [reply src/neon/how-tos/advance-analysis/undistort.md:19]: Then we would probably want to rename some fields like:

"scene_extrinsics_affine_matrix",

**Comment by marc-tonsen** [comment]
Yeah, sorry for the unsolicited feedback! 😅 Neal mentioned this in the lead meeting and I was looking through it to see how it differs from the PI version. And then I thought while I am at it I might as well 😄

**Pull request merged**